examples/ensemble/plot_gradient_boosting_regression.py

```diff
@@ -19,38 +25,79 @@
+from sklearn.model_selection import train_test_split
+
+##############################################################################
+# Load the data
+# -------------------------------------
+#
+# First we need to load the data. We set random state to be consistent with the
+# result.
+
+cancer = datasets.load_breast_cancer()
+X, y = cancer.data, cancer.target
+
+##############################################################################
+# Data preprocessing
+# -------------------------------------
+#
+# Next, we will split our dataset to use 90% for training and leave the rest
+# for testing. We will also prepare the parameters we want to use to fit our
+# regression model. You can play with those parameters to see how the
+# results change:
+#
+# Here:
+# n_estimators : is the number of boosting stages which will be performed.
+#     Later, we will plot and see how the deviance changes with those boosting
+#     operations.
+# max_depth : this limits the number of nodes in the tree. The best value
+#     depends on the interaction of the input variables.
+# min_samples_split : is the minimum number of samples required to split an
+#     internal node.
+# learning_rate: tells how much the contribution of each tree will shrink
+# loss: here, we decided to use least squeares as a loss function, however
+#     there are many other options (check
+#     :class:`~sklearn.ensemble.GradientBoostingRegressor` to see what are
+#     other possibilities)
+
+X_train, X_test, y_train, y_test = train_test_split(X, y,
+                                                    test_size=0.1,
+                                                    random_state=13)
 
-# #############################################################################
-# Load data
-boston = datasets.load_boston()
-X, y = shuffle(boston.data, boston.target, random_state=13)
-X = X.astype(np.float32)
-offset = int(X.shape[0] * 0.9)
-X_train, y_train = X[:offset], y[:offset]
-X_test, y_test = X[offset:], y[offset:]
+params = {'n_estimators': 500,
+          'max_depth': 4,
+          'min_samples_split': 5,
+          'learning_rate': 0.01,
+          'loss': 'ls'}
 
-# #############################################################################
+##############################################################################
 # Fit regression model
-params = {'n_estimators': 500, 'max_depth': 4, 'min_samples_split': 2,
-          'learning_rate': 0.01, 'loss': 'ls'}
-clf = ensemble.GradientBoostingRegressor(**params)
+# -------------------------------------
+#
+# Now we will initiate the gradient boosting regressors and fit it with our
+# training data. Let's also look and the mean squared error on the test data.
 
+clf = ensemble.GradientBoostingRegressor(**params)
 clf.fit(X_train, y_train)
+
 mse = mean_squared_error(y_test, clf.predict(X_test))
-print("MSE: %.4f" % mse)
+print("The mean squared error (MSE) on test set: {:.4f}".format(mse))
```

Comment:
Computing the MSE loss on a classification task is misleading. We should rather use accuracy of ROC AUC. The data is approximately balanced: 62% for the positive class.